packages/provider-utils/src/secure-json-parse.ts

```diff
@@ -0,0 +1,90 @@
+// Licensed under BSD-3-Clause (this file only)
+// Code adapted from https://github.com/fastify/secure-json-parse/blob/783fcb1b5434709466759847cec974381939673a/index.js
+//
+// Copyright (c) Vercel, Inc. (https://vercel.com)
+// Copyright (c) 2019 The Fastify Team
+// Copyright (c) 2019, Sideway Inc, and project contributors
+// All rights reserved.
+//
+// The complete list of contributors can be found at:
+// - https://github.com/hapijs/bourne/graphs/contributors
+// - https://github.com/fastify/secure-json-parse/graphs/contributors
+// - https://github.com/vercel/ai/commits/main/packages/provider-utils/src/secure-parse-json.ts
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+const suspectProtoRx = /"__proto__"\s*:/;
+const suspectConstructorRx = /"constructor"\s*:/;
+
+function _parse(text: string) {
+  // utf8 BOM checker (https://github.com/fastify/secure-json-parse/pull/5)
+  if (text && text.charCodeAt(0) === 0xfeff) {
+    text = text.slice(1);
+  }
```

Comment:
I'm not sure if this is relevant in the context of `@ai-sdk/provider-utils`

Reply:
Seems worth keeping, people might be using proxies.

Reply:
you're right, lets remove this.